drivers/hmis/app/models/hmis/user.rb

```diff
@@ -65,10 +65,15 @@ def load_effective_permissions
       raise "Invalid entity '#{entity.class.name}'" unless joins_association.present?
       return false unless send("#{permission}?")
 
+      access_group_ids = Hmis::GroupViewableEntity.includes_project(entity).pluck(:access_group_id) if entity.is_a?(Hmis::Hud::Project)
+      access_group_ids = Hmis::GroupViewableEntity.includes_organization(entity).pluck(:access_group_id) if entity.is_a?(Hmis::Hud::Organization)
+      access_group_ids = Hmis::GroupViewableEntity.includes_data_source(entity).pluck(:access_group_id) if entity.is_a?(GrdaWarehouse::DataSource)
+      access_group_ids = Hmis::GroupViewableEntity.includes_project_access_group(entity).pluck(:access_group_id) if entity.is_a?(GrdaWarehouse::ProjectAccessGroup)
+
       access_groups.
         merge(
           Hmis::AccessGroup.joins(:access_controls).
-            where(id: Hmis::GroupViewableEntity.where(entity_type: entity.class.name, entity_id: entity.id).pluck(:access_group_id)).
+            where(id: access_group_ids).
             merge(Hmis::AccessControl.where(role_id: roles.where(permission => true).pluck(:id))),
         )
         .exists?
```

Comment:
@ZachCMP for my own understanding here, can you explain why this nested merging approach is preferable to just using a `where` clause? I think they have the same result, but I could be wrong.
```
access_group_ids = ...
role_ids = roles.where(permission => true).pluck(:id)

access_controls.where(access_group_id: access_group_ids, role_id: role_ids).exists?
```

Reply:
@gigxz I changed the code to what you're suggesting here and it seems to work fine. I worked up the initial code with @eanders and followed his lead on how to assemble the query logic, so I probably couldn't explain any better than him why the merging would be needed in this case.

Since the code you suggest is cleaner and also makes the tests pass, I've gone ahead and switched to that.